### PR TITLE
Add TKE budget variables to GradientFlux().

### PR DESCRIPTION
**Pull request by ilopezgp**
### Description

This PR adds TKE budget variables to the GradientFlux() variables of the EDMF scheme. Adding these variables to a state is necessary if we want to output them with the current version of the model.

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

File: test/Atmos/EDMF/edmf_kernels.jl [hunk 1]
```diff
@@ -151,7 +151,12 @@ function vars_state(m::EDMF, st::GradientFlux, FT)
     @vars(
         S²::FT, # should be conditionally grabbed from atmos.turbulence
         environment::vars_state(m.environment, st, FT),
-        updraft::vars_state(m.updraft, st, FT)
+        updraft::vars_state(m.updraft, st, FT),
```
Comment (yairchn) [inline test/Atmos/EDMF/edmf_kernels.jl:154]: is there a reason not to add these variables in the environment in 

https://github.com/CliMA/ClimateMachine.jl/blob/8e2ef08353183dbf6daaf3e5e12d3f5b5a1db0f9/test/Atmos/EDMF/edmf_kernels.jl#L135

File: test/Atmos/EDMF/stable_bl_edmf.jl [hunk 2]
```diff
@@ -198,7 +200,7 @@ function main(::Type{FT}) where {FT}
     grid = driver_config.grid
 
     # state_types = (Prognostic(), Auxiliary(), GradientFlux())
-    state_types = (Prognostic(), Auxiliary())
+    state_types = (Prognostic(), Auxiliary(), GradientFlux())
```
Comment (yairchn) [inline test/Atmos/EDMF/stable_bl_edmf.jl:203]: if we are exporting GradientFlux and entrainment and detrainment are in the output could we compute all the needed values offline ?

File: test/Atmos/EDMF/edmf_kernels.jl (hunk 1 above)
Comment (ilopezgp) [reply test/Atmos/EDMF/edmf_kernels.jl:154]: Good point, I will add them to the environment

File: test/Atmos/EDMF/stable_bl_edmf.jl (hunk 2 above)
Comment (ilopezgp) [reply test/Atmos/EDMF/stable_bl_edmf.jl:203]: Well yes but I would have to rewrite the EDMF offline, I don't see the point if I can just get the output here. The same is applicable to entrainment and detrainment.

Comment (ilopezgp) [reply test/Atmos/EDMF/stable_bl_edmf.jl:203]: The ideal think would be if I could get the `atmos_source!` variables

**Review by charleskawczynski — APPROVED** [review]
I'm fine with these changes assuming that this will help edmf development. Adding fields to `vars_state(::GradientFlux)` allocates more global memory, which we'll need to eventually remove. So, it'll behoove us to avoid depending on these fields inside `flux_first_order!`/`source!` since we'll need to eventually remove them.

**Comment by ilopezgp** [comment]
> I'm fine with these changes assuming that this will help edmf development. Adding fields to `vars_state(::GradientFlux)` allocates more global memory, which we'll need to eventually remove. So, it'll behoove us to avoid depending on these fields inside `flux_first_order!`/`source!` since we'll need to eventually remove them.

I agree! We will have to remove it at some point, this is just temporary.

**Pull request merged**